fili-core/src/main/java/com/yahoo/bard/webservice/application/AbstractBinderFactory.java

```diff
@@ -919,7 +924,24 @@ protected JobPayloadBuilder buildJobPayloadBuilder() {
      * @return A DruidWebService
      */
     protected DruidWebService buildDruidWebService(DruidServiceConfig druidServiceConfig, ObjectMapper mapper) {
-        return new AsyncDruidWebServiceImpl(druidServiceConfig, mapper, HashMap::new);
+        Supplier<Map<String, String>> supplier = HashMap::new;
+        String customSupplierClassString = SYSTEM_CONFIG.getStringProperty(DRUID_HEADER_SUPPLIER_CLASS, null);
+        if (customSupplierClassString != null && customSupplierClassString != "") {
+            try {
+                Class<?> c = Class.forName(customSupplierClassString);
+                Constructor<?> constructor = c.getConstructor();
+                supplier = (Supplier<Map<String, String>>) constructor.newInstance();
+            } catch (ClassNotFoundException | NoSuchMethodException |
+                    IllegalAccessException | InstantiationException | InvocationTargetException e) {
```

Comment:
Rather than enumerate all of these, should we just catch `Exception`? If anything goes wrong loading this, we should just fail anyway, no matter what the exception is, right?